index.js

```diff
@@ -358,6 +358,12 @@ module.exports = class Hyperdrive extends ReadyResource {
     await Promise.all(proms)
   }
 
+  async rename (oldPath, newPath, opts) {
+    const existingEntry = await this.entry(oldPath, opts)
+    await this.db.del(std(oldPath, false), { keyEncoding })
+    return this.db.put(std(newPath, false), existingEntry.value, { keyEncoding })
```

Comment:
This logic is prone to race conditions:
- If the process crashes/stops after doing the del, but before the put runs, the entry's metadata no longer exists
- if rename runs twice for the same oldPath and newPath, and the second one starts after the first one finished its del operation (but before it finished its put), then `existingEntry` will be null, so the second run will override the result of the first one and make it point to null

The first issue can be resolved by using a batch, and the second with some sort of lock or debounce, but I'm not sure it's worth the complexity for this feature

Reply:
Great catch! I should have thought of this.
Added `drive.batch()` to lock the drive while renaming which should address the race condition and the mid process crash concern.